Commande /vocal

Asthox opened the pull request.

Asthox committed:
Commande /vocal

Asthox committed:
Adaptation discordjs V13 avec Permissions.FLAGS

Asthox committed:
Adaptation discordjs V13 avec Permissions.FLAGS

Asthox committed:
Ajout permission

Asthox committed:
Ajustement texte

TanguyChiffoleau committed:
Utilisation "Constants.ChannelTypes"

TanguyChiffoleau committed:
Condition filtre plus explicite

TanguyChiffoleau committed:
Ajouts espaces pour aérer

TanguyChiffoleau committed:
Suppression check du channel inutile

TanguyChiffoleau committed:
Suppression check inutile car option required

TanguyChiffoleau committed:
Réponse avec nb user dans channel retourné par API

Asthox committed:
Merge branch 'TanguyChiffoleau:master' into Commande-vocal

Asthox committed:
Setup des permissions correctement avec set() + edit()

Asthox committed:
Utilisation de Constants + catch erreur

Le catch erreur permet d'éviter le plantage au cas où la personne clique sur un autre salon entre temps, car le interaction.reply() met un peu de temps (le temps de créer le salon et setup les perms)

Asthox committed:
Merge branch 'TanguyChiffoleau:master' into Commande-vocal

Asthox committed:
Merge branch 'TanguyChiffoleau:master' into Commande-vocal

Asthox committed:
Suppression permission déjà refusée

Asthox committed:
Retouche du système de permissions

TanguyChiffoleau committed:
Ajout constante pour overwriter member

TanguyChiffoleau commented on src/events/voice/voiceStateUpdate.js:
```diff
@@ -65,7 +69,6 @@ const handleJoin = async (newState, client) => {
 	if (noMicChannel)
 		// On lui donne la permission de voir le salon
 		return noMicChannel.permissionOverwrites.edit(newState.id, {
-			CREATE_INSTANT_INVITE: false,
```
Pourquoi avoir supprimé cette permission ? Elle permet de pouvoir créer un lien d'invitation directement vers le vocal, si la personne y a accès

Asthox replied:
https://github.com/TanguyChiffoleau/Le-bot-en-JS/pull/258/files/46cb4fc2ea5f085ff0dbc8580eb2e64eef2ae40b#diff-bedc7c244b885f1529ee4d30af1ef741e6c600ade1c91649c9b546ceebdcd161R83

Car je la met en deny directement lors de la création du salon

Du coup lorsqu'on edite, on ne touche pas à cette perm, inutile de la refuser 2 fois

TanguyChiffoleau committed:
Suppression et fix des eslint-disable

Asthox committed:
Merge branch 'TanguyChiffoleau:master' into Commande-vocal

TanguyChiffoleau committed:
Fix et suppression eslint-disable-next-line no-unused-expressions

TanguyChiffoleau commented on src/slash-commands/custom-voice-channels/vocal.js:
```diff
@@ -0,0 +1,156 @@
+						content: `Ton salon a bien été créé : ${noMicChannel} 👌`,
+						ephemeral: true,
+					})
+					.catch(() => null)
```
Quelle(s) erreur(s) peu(ven)t être(s) générée(s) ? Il faudrait les gérer proprement plutôt que de toujours renvoyer null

Asthox replied:
Le null a été mis car parfois si la création prend du temps, l'interaction répond "The application did not respond", mais ne génère aucune erreur
ça permettait de ne pas afficher cette réponse

TanguyChiffoleau replied:
Si parfois c'est trop long, on peut utiliser `interaction.deferReply()` pour étendre la validié du token et éviter ça : https://discordjs.guide/interactions/replying-to-slash-commands.html#deferred-responses

Asthox replied:
https://github.com/TanguyChiffoleau/Le-bot-en-JS/pull/258/commits/a9ecd101a2b8a81a622dbd67a9f9d56b648e705f

TanguyChiffoleau committed:
Merge branch 'Commande-vocal' of https://github.com/asthox/userbot into pr/Asthox/258

Asthox committed:
deferReply() afin d'étendre la validité du token d'interaction

https://github.com/TanguyChiffoleau/Le-bot-en-JS/pull/258#discussion_r810489998

Asthox committed:
Merge branch 'TanguyChiffoleau:master' into Commande-vocal

Asthox committed:
Merge branch 'TanguyChiffoleau:master' into Commande-vocal

Asthox committed:
Merge branch 'TanguyChiffoleau:master' into Commande-vocal

Asthox committed:
Merge branch 'TanguyChiffoleau:master' into Commande-vocal

Asthox committed:
Merge branch 'TanguyChiffoleau:master' into Commande-vocal

Asthox committed:
Merge branch 'TanguyChiffoleau:master' into Commande-vocal

Asthox committed:
Ajout couleurs noms salons

Asthox committed:
Merge branch 'TanguyChiffoleau:master' into Commande-vocal

Asthox committed:
Merge branch 'TanguyChiffoleau:master' into Commande-vocal

Asthox committed:
Merge branch 'TanguyChiffoleau:master' into Commande-vocal
